### ISSUE TITLE
Bower package of React-Bootstrap has been moved to separate repo which makes AMD paths unusable

### ISSUE THREAD

**Issue by kirill-konshin** [react-bootstrap/react-router-bootstrap#53]
AMD paths in build:

```js
define(["react", "react-bootstrap/lib/Button", "react-bootstrap/lib/ListGroupItem", "react-bootstrap/lib/MenuItem", "react-bootstrap/lib/NavItem"], factory);
```

Should now be:

```js
define(["react", "react-bootstrap/Button", "react-bootstrap/ListGroupItem", "react-bootstrap/MenuItem", "react-bootstrap/NavItem"], factory);
```

(without `/lib/`)

To do that in `webpack.config.js` sections like 

```
amd: "react-bootstrap/lib/Button"
```

Should now be

```
amd: "react-bootstrap/Button"
```

**Comment by kirill-konshin** [comment]
PR https://github.com/react-bootstrap/react-router-bootstrap/pull/54

**Comment by taion** [comment]
Is this the right thing to do, rather than to change the AMD build to also use the same paths? It might be nice to be consistent across environments.

**Comment by kirill-konshin** [comment]
I think the answer is yes, it's better to keep paths consistent. I'll create another ticket.

**Comment by kirill-konshin** [comment]
https://github.com/react-bootstrap/react-bootstrap/issues/679 created

**Comment by kirill-konshin** [comment]
Actually, the issue is much more complex...

React-Router-Bootstrap requires individual files from React-Bootstrap, and those files require `react/lib/ReactTransitionEvents`. Which obviously is not accessible if React is installed via Bower because React neither exposes `ReactTransitionEvents` nor has the file as a part of the package.

I think this is the reason why the build of React-Bootstrap contains some modified version of `ReactTransitionEvents`. Unfortunately the build cannot be used because individual files are listed as dependencies...

I totally understand that it has been done in order to reduce the size of the overall build when using Browserify/Webpack, but this makes React-Bootstrap/React-Router-Bootstrap completely unusable with RequireJS.

**Comment by kirill-konshin** [comment]
Some more stuff that I've found:

1. `react-bootstrap/lib/ModalTrigger` and `react-bootstrap/lib/OverlayTrigger` are not listed as external dependencies
2. The use of `ModalTrigger.withContext` fails since `withContext` is deprecated

**Comment by taion** [comment]
How did `withContext` get deprecated? I just added that feature very recently.

**Comment by kirill-konshin** [comment]
React source:

```js
  withContext: function(newContext, scopedCallback) {
    if ("production" !== "development") {
      ("production" !== "development" ? warning(
        didWarn,
        'withContext is deprecated and will be removed in a future version. ' +
        'Use a wrapper component with getChildContext instead.'
      ) : null);

      didWarn = true;
    }
```

Article: https://www.tildedave.com/2014/11/15/introduction-to-contexts-in-react-js.html (look at Update (2/19/2015)).

**Comment by taion** [comment]
That's not the method being used here. I may have chosen a poor name, but it's actually a separate one I'm defining here: https://github.com/react-bootstrap/react-bootstrap/blob/master/src/OverlayTrigger.js#L314

**Comment by kirill-konshin** [comment]
Whoa! But still the problem I faced: this method was undefined... I have no idea why...

**Comment by kirill-konshin** [comment]
I will double check the undefined method once everything from this comment will be addressed: https://github.com/react-bootstrap/react-bootstrap/pull/682#issuecomment-102667774

**Comment by taion** [comment]
@kirill-konshin FWIW, it was just added with the latest react-bootstrap release. I believe I updated the `bower.json` dependencies accordingly, but it may have been an issue with my having forgotten to add those modules to the configured externals.

**Comment by kirill-konshin** [comment]
So far I don't see any changes neither in `bower.json` nor in directory layout...

**Comment by taion** [comment]
We need to make a new release first before these changes will show up.

**Comment by kirill-konshin** [comment]
How long is it going to take?

**Comment by taion** [comment]
Should be available in the new 0.22.4 now.

**Comment by kirill-konshin** [comment]
Great, looks like it's now working when I put a package in RJS config:

```
{
    packages: [
        {
            name: 'react-bootstrap',
            location: 'bower_components/react-bootstrap',
            main: 'lib/index'
        }
    ]
}
```

So the last thing left is to build router-bootstrap...

**Comment by kirill-konshin** [comment]
I've tested the latest changes, only one thing left -- to publish release of Router-Bootstrap. When to expect this?

**Comment by taion** [comment]
I just published v0.15.0.

**Comment by kirill-konshin** [comment]
OK, everything seems to be good.
